HW22/HW22.cpp

```diff
@@ -0,0 +1,160 @@
+﻿#include <iostream>
+#include "DynamicArray.h"
+
+template<typename T>
+DynamicArray<T>& DynamicArray<T>::operator=(const DynamicArray& other)
+{
+    if (this != &other)
+    {
+        delete[] data;
+        copyData(other);
+    }
+    return *this;
+}
+
+template<typename T>
+T& DynamicArray<T>::operator[](std::size_t index)
+{
+
+    return data[index];
+}
+
+template<typename T>
+void DynamicArray<T>::setSize(std::size_t newSize)
+{
+    if (newSize <= capacity)
+    {
+        size = newSize;
+    }
+    else
+    {
+        reserve(newSize);
+        size = newSize;
+    }
+}
+
+template<typename T>
+std::size_t DynamicArray<T>::getSize() const
+{
+    return size;
+}
+
+template<typename T>
+void DynamicArray<T>::clear()
+{
+    delete[] data;
+    data = nullptr;
+    size = 0;
+    capacity = 0;
+}
+
+template<typename T>
+void DynamicArray<T>::push_back(T element)
+{
+    if (size == capacity)
+    {
+        reserve(capacity * 2 + 1);
+    }
+    data[size++] = element;
+}
+
+
+
+template<typename T>
+void DynamicArray<T>::pop_back()
+{
+    if (size > 0)
+    {
+        --size;
+    }
+}
```

Comment:
- Create array with 1000 elements
- call pop_back() 999 times
- the size will be 1, but the array will still occupy 1000*(sizeof(T)) bytes? Not quite right)